plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/completion/SQLCompletionAnalyzer.java

```diff
@@ -1622,24 +1640,28 @@ public void visit(@Nullable CreateView createView) {
             params);
     }
     
-    @NotNull
-    private String getAliceFromQueryANTLR(
-        @NotNull String tableName,
-        @NotNull String query) throws IOException {
-        String alias = "";
+    /**
+     * The method returns table to alias map
+     */
+    @Nullable
+    private Map<String, String> getTableAlicesFromQueryByANTLRParser(
```

Comment:
We have LSMTableReferencesAnalyzer interface with 2 implementations. One implementation contains method for getting table references from syntax tree built by antlr parser, another one is previously used analyzer. And implementation is regulated by the setting Enable experimental feature for autocompletion. Let's keep this idea and move this method to one of the realization of this interface (TableReferencesAnalyzer). Method for OldTableReferencesAnalyzer could be empty if there was nothing previously for this case. Also it will allow to turn this feature off if something goes wrong with parser.